### PR TITLE
feat(core): Add FsModuleLoader that supports loading from filesystem

### PR DESCRIPTION
**Pull request by magurotuna**
This commit adds `FsModuleLoader` to `deno_core`, which implements `ModuleLoader` trait.
It is used when creating a runtime that supports module loading from filesystem. In particular, it's going to be used by `deno_lint` plugin feature. 

Ref: 
- https://github.com/denoland/deno_lint/pull/415
- https://github.com/denoland/deno/pull/8381

CC @bartlomieju

### CODE REVIEW

File: core/modules.rs
Comment (rheh) [inline core/modules.rs:159]: Could these three let be const?

File: core/modules.rs
Comment (rheh) [inline core/modules.rs:156]: why async?  I see no await within?

File: core/modules.rs
Comment (rheh) [inline core/modules.rs:158]: Renamed to content -> code?

File: core/modules.rs
Comment (rheh) [inline core/modules.rs:160]: If we renamed to code, we could just:
`{
code,
...
}`

File: core/modules.rs
Comment (bartlomieju) [reply core/modules.rs:159]: The trait requires to return future from this method.

File: core/modules.rs
Comment (bartlomieju) [reply core/modules.rs:167]: No, Rust doesn't allow that

**Comment by bartlomieju** [comment]
@magurotuna I applied some changes to error handling and doc comment. I'll wait on @ry's review before landing

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**
